### PR TITLE
[API/Single] Elaborate error logs of ml_single_invoke

### PR DESCRIPTION
**Pull request by anyj0527**
Add error logs for input tensor size mismatching cases.
Please give me some feedback!

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

Resolves (partially) : #2127 

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: api/capi/src/nnstreamer-capi-single.c [hunk 1]
```diff
@@ -748,32 +758,49 @@ ml_single_invoke (ml_single_h single,
...
+    status = ML_ERROR_TRY_AGAIN;
+    goto exit;
+  }
+
   /* Validate input data */
   if (in_data->num_tensors != single_h->in_info.num_tensors) {
-    ml_loge ("The given param input is invalid, \
-        different number of memory blocks.");
+    ml_loge ("The number of input tensors is not compatible with model. \
+        Given: %u, Expeceted: %u.", in_data->num_tensors,
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single.c]: typo : Expeceted

Comment (anyj0527) [reply api/capi/src/nnstreamer-capi-single.c]: Thank you! I fixed it.

File: api/capi/src/nnstreamer-capi-single.c [hunk 2]
```diff
@@ -748,32 +758,49 @@ ml_single_invoke (ml_single_h single,
...
 
-  if (!single_h->filter || single_h->state >= JOIN_REQUESTED) {
+  if (!single_h->filter) {
     ml_loge ("The given param is invalid, model is missing.");
     status = ML_ERROR_INVALID_PARAMETER;
     goto exit;
   }
 
+  if (single_h->state >= JOIN_REQUESTED) {
+    ml_loge ("The thread is not idle.");
```
Comment (myungjoo) [inline api/capi/src/nnstreamer-capi-single.c]: Incorrect message. This means that the handle is closed or being closed. (refer to nnstreamer-capi-single.c:88)

File: api/capi/src/nnstreamer-capi-single.c [hunk 3]
```diff
@@ -748,32 +758,49 @@ ml_single_invoke (ml_single_h single,
   in_data = (ml_tensors_data_s *) input;
   *output = NULL;
 
-  if (!single_h->filter || single_h->state >= JOIN_REQUESTED) {
+  if (!single_h->filter) {
```
Comment (myungjoo) [inline api/capi/src/nnstreamer-capi-single.c]: You need to understand what's going on with this code and rewrite the error message. The current error message doesn't look accurate.

File: api/capi/src/nnstreamer-capi-single.c [hunk 4]
```diff
@@ -748,32 +758,49 @@ ml_single_invoke (ml_single_h single,
...
+      ml_loge ("The size of %d-th input tensor is not compatible with model. \
+          Given: %zu, Expected: %zu (type: %d).",
+          i, in_data->tensors[i].size, raw_size, single_h->in_info.info[i].type);
       status = ML_ERROR_INVALID_PARAMETER;
       goto exit;
     }
   }
 
   if (single_h->state != IDLE) {
+    ml_loge ("The thread is not idle.");
```
Comment (myungjoo) [inline api/capi/src/nnstreamer-capi-single.c]: You need to be more specific: users will wonder which thread you are talking about..

File: api/capi/src/nnstreamer-capi-single.c [hunk 5]
```diff
@@ -748,32 +758,49 @@ ml_single_invoke (ml_single_h single,
...
-  if (!single_h->filter || single_h->state >= JOIN_REQUESTED) {
+  if (!single_h->filter) {
     ml_loge ("The given param is invalid, model is missing.");
     status = ML_ERROR_INVALID_PARAMETER;
     goto exit;
   }
 
+  if (single_h->state >= JOIN_REQUESTED) {
+    ml_loge ("The thread is not idle.");
+    status = ML_ERROR_TRY_AGAIN;
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single.c]: plz return 'ML_ERROR_STREAMS_PIPE' here

File: api/capi/src/nnstreamer-capi-single.c [hunk 6]
```diff
@@ -738,8 +738,18 @@ ml_single_invoke (ml_single_h single,
...
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!input) {
+    ml_loge ("The second param, input is not valid.");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!output) {
+    ml_loge ("The third param, output is not valid.");
```
Comment (wooksong) [inline api/capi/src/nnstreamer-capi-single.c]: IMO, messages indicating the first, second, and third parameters are invalid are hardly helpful for the application developers. Although the previous message is also about the wrong parmaters, you need to improve it.

**Comment by anyj0527** [comment]
@myungjoo @jaeyun-jung @wooksong 
I updated this commit based on your feedback. Please review the modifications.

File: api/capi/src/nnstreamer-capi-single.c [hunk 7]
```diff
@@ -738,8 +738,21 @@ ml_single_invoke (ml_single_h single,
 
   check_feature_state ();
 
-  if (!single || !input || !output) {
-    ml_loge ("The given param is invalid.");
+  if (!single) {
+    ml_loge ("The first argument of ml_single_invoke() is not valid. \
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single.c]: You don't need to divide log sring. Write it one line. (no limit in column)

**Comment by anyj0527** [comment]
@myungjoo @jaeyun-jung , Thank you for reviewing about the string convention
PR updated:
Do not make line break in printf string.

**Pull request merged**